.github/workflows/tests.yaml

```diff
@@ -0,0 +1,139 @@
+name: Unit and E2E Tests
+on: [pull_request]
```

Comment:
since we are dependent on external services here I think it would be useful to run e2e tests also once a week and also on pull requests. example: https://github.com/logzio/logzio_terraform_client/blob/master/.github/workflows/test-daily.yml#L4